src/CLClient.ts

```diff
@@ -271,7 +271,7 @@ export class CLClient {
     public TrainDialogValidateTextVariation(appId: string, trainDialogId: string, textVariation: CLM.TextVariation, filteredDialog: string): Promise<null> {
         let apiPath = `app/${appId}/traindialog/${trainDialogId}/extractor/textvariation`
         // Note: service can take a list of filteredDialogs, but we just use one for now
-        let query = filteredDialog ? `filteredDialogs[]=${filteredDialog}` : undefined
+        let query = filteredDialog ? `filteredDialogs=${filteredDialog}` : undefined
```

Comment:
This is so confusing.

`filteredDialog` Implied it's an object but yet it's type is a string so we assume it's a id.  Either way it's a single thing.
Then we assign it to a `filteredDialogs` which is plural

Does that mean the API does allow taking multiple id's?
This would imply the fuction should also allow passing multiple ids so it can construct the proper query parameter

If it doesn't take multiple ids then why is it plural?

Reply:
Shahin wrote the API to take multiple ids and called it "filteredDialogs".  Currently we only have a use case for one

Reply:
Then it would make sense to pass an array with one id instead of changing the API.
Can keep it required by using tuples which are open ended. `[string]`

Stuff like this just adds to all the things we have to remember.  2 months later you realize you want to filter another dialog and think oh dang we can't, it only accepts one, then go look at the server to find out it does, and then go find out which functions (this) does the conversion, then update all the types and method signatures just to add one string

Reply:
Shahin and I look at this together last week.  ASP.net (and his code) is taking multiple items in the form of "filteredDialogs={id1}&filteredDialogs={id2}" etc

Reply:
Yes, and how would you pass in `id1`, `id2` and so on to construct that parameter?

I only see `filteredDialog: string`

Did he actually say to change the API on the client to only take 1 id? If so, what was the reason?

The construction and format of the parameter is independent problem to having the all information required to construct it.

Reply:
The former method (filteredDialogs[]=${filteredDialog}) did not work with the server setup.   If / when we find the need to send more than one, we can spend more time on this.  For now we can't think of any reason why this would be the case